docs/source/recipes/authentication.md

```diff
@@ -39,43 +39,58 @@ Note: the backend must also allow credentials from the requested origin. e.g. if
 
 ## Header
 
-Another common way to identify yourself when using HTTP is to send along an authorization header. Apollo Links make creating middlewares that lets you modify requests before they are sent to the server. It's easy to add an `Authorization` header to every HTTP request. In this example, we'll pull the login token from `localStorage` every time a request is sent:
+Another common way to identify yourself when using HTTP is to send along an authorization header. Apollo Links make creating middlewares that lets you modify requests before they are sent to the server. It's easy to add an `Authorization` header to every HTTP request. In this example, we'll pull the login token from `localStorage` every time a request is sent.
+
+In `graphql.module.ts`:
 
 ```ts
-import { HttpHeaders } from '@angular/common/http';
-import { Apollo } from 'apollo-angular';
-import { HttpLink } from 'apollo-angular-link-http';
+import { NgModule } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
+import { ApolloModule, Apollo, APOLLO_OPTIONS } from 'apollo-angular';
+import { HttpLinkModule, HttpLink } from 'apollo-angular-link-http';
 import { InMemoryCache } from 'apollo-cache-inmemory';
+import { ApolloLink } from 'apollo-link';
 
-@NgModule({ ... })
-class AppModule {
-  constructor(
-    apollo: Apollo,
-    httpLink: HttpLink
-  ) {
-    const http = httpLink.create({uri: '/graphql'});
+const uri = '/graphql';
 
-    const authLink = new ApolloLink((operation, forward) => {
-      // Get the authentication token from local storage if it exists
-      const token = localStorage.getItem('token');
+export function provideApollo(httpLink: HttpLink) {
+  const basic = setContext((operation, context) => ({
+    headers: {
+      Accept: 'charset=utf-8'
```

Comment:
@herkulano @kamilkisiela This doesn't look like a valid `Accept` header: it doesn't contain a MIME type like `foo/bar`. Does this example actually work well in practice, or should this be updated to `*/*; charset=utf-8` or something?